### PR TITLE
ssize_t is not defined in GdbStub.h or GdbCmds.h

### PR DESCRIPTION
**Pull request by RealAstolfo**
i had to add sys/types.h and patch my gentoo ebuilds for a successful compilation.

### CODE REVIEW

**Comment by RSDuck** [comment]
https://en.cppreference.com/w/cpp/types/size_t

> Defined in header [<cstddef>](https://en.cppreference.com/w/cpp/header/cstddef)

?

**Comment by CasualPokePlayer** [comment]
ssize_t is the non-standard signed variant of size_t

**Comment by RealAstolfo** [comment]
> ssize_t is the non-standard signed variant of size_t. It would typically also only be defined on Unix-y platforms (on Windows you'd have to define it yourself, probably to intptr_t since that's the same in that context)

exactly this. only POSIX systems have ssize_t, the windows equivelent is actually SSIZE_T



> https://en.cppreference.com/w/cpp/types/size_t
> 
> > Defined in header [<cstddef>](https://en.cppreference.com/w/cpp/header/cstddef)
> 
> ?

as for this, that is C++, this is a C header. including <cstddef> in a C file is a good way to make the compiler switch to C++ mode imo

**Comment by RealAstolfo** [comment]
i just ammended the changes to something more appropriate for C++, which should still fix the cited build issues on ssize_t on my platform

File: src/debug/GdbCmds.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-
+#include <cstddef>
```
Comment (CasualPokePlayer) [inline src/debug/GdbCmds.h:1]: GdbCmds.h is only every supposed to be included in gdb stub, so the include here is pointless.

Comment (RealAstolfo) [reply src/debug/GdbCmds.h]: got it. ill remove that part of the patch (on the other hand, what a weird way to include code)

**Comment by RealAstolfo** [comment]
Seems like the patch should be finalized, so heres the review.

What i did: 
added `#include <sys/types.h>` to src/debug/GdbStub.h

Why:
on this gentoo system (specifically the musl libc), the program would fail to compile with errors saying ssize_t is not declared/defined.

What it accomplished:
with -DGDB_ENABLE, the gentoo system now successfully compiles the program,
and now the program is more compliant to posix.

**Pull request merged**